pkg/detectors/buildkite/v1/buildkite.go

```diff
@@ -79,3 +72,24 @@ func (s Scanner) Type() detectorspb.DetectorType {
 func (s Scanner) Description() string {
 	return "Buildkite is a platform for running fast, secure, and scalable continuous integration pipelines. Buildkite API tokens can be used to access and modify pipeline data and configurations."
 }
+
+func VerifyBuildKite(ctx context.Context, client *http.Client, secret string) (bool, error) {
+	// create a request
+	req, err := http.NewRequestWithContext(ctx, "GET", "https://api.buildkite.com/v2/access-token", nil)
+	if err != nil {
+		return false, err
+	}
+
+	// add authorization header
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", secret))
+
+	res, err := client.Do(req)
+	if err == nil {
+		defer res.Body.Close()
+		if res.StatusCode >= 200 && res.StatusCode < 300 {
+			return true, nil
+		}
+	}
```

Comment:
This old pattern is pretty lax. A better implementation is to (i) check the error, (ii) defer the drain + close of the body, (iii) check for a specific status code.

https://github.com/trufflesecurity/trufflehog/blob/a6e2b9995d5e6f9db9d7dc3430fcfed04f6688fb/pkg/detectors/alchemy/alchemy.go#L78-L81